modules/sdk-coin-ada/src/lib/transaction.ts

```diff
@@ -152,6 +206,21 @@ export class Transaction extends BaseTransaction {
       this._transaction = txn;
       this._id = Buffer.from(CardanoWasm.hash_transaction(txn.body()).to_bytes()).toString('hex');
       this._type = TransactionType.Send;
+      if (this._transaction.body().certs()) {
+        for (let i = 0; i < this._transaction.body().certs()!.len(); i++) {
+          const cert = this._transaction.body().certs()!.get(i);
+          if (cert.as_stake_registration() !== undefined) {
+            this._type = TransactionType.StakingActivate;
```

Comment:
Is there value in `break`ing the loop once `this_type` is assigned?

Reply:
meh, not really imo, certs array will only be like 2 certs long max